server/service/src/sync/translations/master_list.rs

```diff
@@ -59,6 +129,94 @@ impl SyncTranslation for MasterListTranslation {
     }
 }
 
+struct GenerateRequisitionProgram {
+    pub program_row: ProgramRow,
+    pub program_requisition_settings_row: ProgramRequisitionSettingsRow,
+    pub program_requisition_order_type_rows: Vec<ProgramRequisitionOrderTypeRow>,
+}
+
+fn generate_requisition_program(
+    connection: &StorageConnection,
+    master_list: &LegacyListMasterRow,
+) -> Result<Option<GenerateRequisitionProgram>, anyhow::Error> {
+    if master_list.is_program == false {
+        return Ok(None);
+    }
+
+    let program_settings = master_list.program_settings.clone().ok_or(anyhow::anyhow!(
+        "Program settings not found for program {}",
+        master_list.id
+    ))?;
+    let order_types = program_settings
+        .store_tags
+        .tags
+        .values()
+        .filter_map(|tag| tag.order_types.clone())
+        .flatten()
+        .collect::<Vec<LegacyOrderType>>();
+
+    let name_tag_and_period_schedule_ids = get_name_tag_and_period_schedule_id(
+        connection,
+        &program_settings.store_tags,
+        &master_list.id.clone(),
+    )?;
+
+    let program_row = ProgramRow {
+        id: master_list.id.clone(),
+        name: master_list.description.clone(),
+    };
+
+    let program_requisition_settings_row = ProgramRequisitionSettingsRow {
+        // Concatenate the program id and name tag id to create a unique id
+        // instead of using uuid since easier to test this way.
+        id: format!("{}{}", master_list.id, name_tag_and_period_schedule_ids.0),
+        name_tag_id: name_tag_and_period_schedule_ids.0,
+        program_id: master_list.id.clone(),
+        period_schedule_id: name_tag_and_period_schedule_ids.1,
+    };
+
+    let program_requisition_order_type_rows = if !order_types.is_empty() {
```

Comment:
Do we need the `is_empty()` check? won't iter over and empty return an empty vec?